apps/hpc-ftsadmin/src/app/pages/add-edit/flow-edit.tsx

```diff
@@ -0,0 +1,1857 @@
+        );
+      }
+
+      if (flowData.destinationOrganizations.length > 0) {
+        flowsFilterInitialValues.destinationOrganizations = getSearchKeyValues(
+          flowData,
+          'destinationOrganizations'
+        );
+      }
+
+      if (flowData.destinationPlans.length > 0) {
+        flowsFilterInitialValues.destinationPlans = getSearchKeyValues(
+          flowData,
+          'destinationPlans'
+        );
+      }
+      paramsObject['filters'] = JSON.stringify(
+        encodeFilters(flowsFilterInitialValues, FLOWS_FILTER_INITIAL_VALUES)
+      );
+
+      const params = new URLSearchParams(paramsObject);
+      const newUrl = `/flows/?${params.toString()}`;
+      window.open(newUrl, '_blank');
+    },
+    [flowData]
+  );
+
+  const handleRestricted = (isChecked: boolean) => {
+    if (isChecked) {
+      setIsRestricted(true);
+    } else {
+      setIsRestricted(false);
+    }
+  };
+  const handleErrorCorrection = (isChecked: boolean) => {
+    if (isChecked) {
+      setErrorCorrection(true);
+    } else {
+      setErrorCorrection(false);
+    }
+  };
+
+  const initializeInputEntries = () => {
+    const tmpEntries = JSON.parse(JSON.stringify(initialInputEntries));
+    Object.keys(inputEntries).forEach((key) => {
+      const specificKey = key as keyof InputEntriesType;
+      const value = inputEntries[specificKey];
+
+      if (isRight(ts.array(forms.INPUT_ENTRY_TYPE).decode(value))) {
+        tmpEntries[specificKey] = (value as forms.InputEntryType[]).filter(
+          (inputEntry) =>
+            inputEntry.category !== forms.InputEntryCategoriesEnum.ACTIVE_FLOW
+        );
+      }
+    });
+    setPendingFieldsallApplied(true);
+    setInputEntries(tmpEntries);
+  };
+
+  const rejectInputEntry = (key: string) => {
+    const tmpEntries = JSON.parse(JSON.stringify(inputEntries));
+    const specificKey = key as keyof InputEntriesType;
+    const value = inputEntries[specificKey];
+    if (isRight(ts.array(forms.INPUT_ENTRY_TYPE).decode(value))) {
+      tmpEntries[specificKey] = (value as forms.InputEntryType[]).filter(
+        (inputEntry) =>
+          inputEntry.category !== forms.InputEntryCategoriesEnum.ACTIVE_FLOW
+      );
+    } else {
+      tmpEntries[specificKey] = null;
+    }
+
+    setInputEntries(tmpEntries);
+  };
+
+  const createInputEntryActiveFlow = (
+    pendingData: IndividualFormValueType,
+    currentData: IndividualFormValueType,
+    key: keyof InputEntriesType
+  ) => {
+    const inputEntry = {
+      category: forms.InputEntryCategoriesEnum.ACTIVE_FLOW,
+      value: pendingData,
+      kind: !pendingData
+        ? forms.InputEntryKindsEnum.DELETED
+        : !currentData
+        ? forms.InputEntryKindsEnum.NEW
+        : forms.InputEntryKindsEnum.REVISED,
+    } as forms.InputEntryType;
+
+    if (compareSelectValues(pendingData, currentData)) {
+      if (isArrayFieldOfInputEntries(key)) {
+        (inputEntries[key] as forms.InputEntryType[]).push(inputEntry);
```

Comment:
Avoid Type Assertion. Even though you checked before in the if statement, if TypeScript complains it means is not well typed. I would recommend read into Type guards for this.